internal/pkg/admission/policy-server-ca-secret.go

```diff
@@ -73,37 +83,40 @@ func (r *Reconciler) fetchOrInitializePolicyServerCASecret(ctx context.Context,
 	return nil
 }
 
-func extractCaFromSecret(caSecret *corev1.Secret) (*admissionregistration.CA, error) {
+func extractCAFromSecret(caSecret *corev1.Secret) ([]byte, *rsa.PrivateKey, error) {
 	caCert, ok := caSecret.Data[constants.PolicyServerCARootCACert]
 	if !ok {
-		return nil, fmt.Errorf("CA could not be extracted from secret %s", caSecret.Kind)
+		return nil, nil, fmt.Errorf("CA could not be extracted from secret %s", caSecret.Kind)
 	}
-	caPrivateKeyBytes, ok := caSecret.Data[constants.PolicyServerCARootPrivateKeyCertName]
+
+	privateKeyBytes, ok := caSecret.Data[constants.PolicyServerCARootPrivateKeyCertName]
 	if !ok {
-		return nil, fmt.Errorf("CA private key bytes could not be extracted from secret %s", caSecret.Kind)
+		return nil, nil, fmt.Errorf("CA private key bytes could not be extracted from secret %s", caSecret.Kind)
 	}
 
-	caPrivateKey, err := x509.ParsePKCS1PrivateKey(caPrivateKeyBytes)
+	privateKey, err := x509.ParsePKCS1PrivateKey(privateKeyBytes)
 	if err != nil {
-		return nil, fmt.Errorf("CA private key could not be extracted from secret %s", caSecret.Kind)
+		return nil, nil, fmt.Errorf("CA private key could not be extracted from secret %s", caSecret.Kind)
 	}
-	return &admissionregistration.CA{CaCert: caCert, CaPrivateKey: caPrivateKey}, nil
+
+	return caCert, privateKey, nil
 }
 
-func (r *Reconciler) fetchOrInitializePolicyServerCARootSecret(ctx context.Context, policyServer *policiesv1.PolicyServer, generateCA generateCAFunc, pemEncodeCertificate pemEncodeCertificateFunc) (*corev1.Secret, error) {
+func (r *Reconciler) fetchOrInitializePolicyServerCARootSecret(ctx context.Context, policyServer *policiesv1.PolicyServer) (*corev1.Secret, error) {
```

Comment:
Same here, the naming is confusing since no CA is stored inside of the `Secret` that is returned